src/ProxyManager/Inflector/ClassNameInflector.php

```diff
@@ -37,14 +38,14 @@ public function getUserClassName(string $className) : string
         $className = ltrim($className, '\\');
         $position  = strrpos($className, $this->proxyMarker);
 
-        if ($position === false) {
+        if (! is_int($position)) {
```

Comment:
is this change actually needed by psalm ? It should be able to detect that the other case has an int when comparing to false, as the signature is `int|false`.

Reply:
The inference for now is `int|bool`, not `int|false`. I'm fine with adapting the code for these scenarios for now, as upstream work for PHPStorm stubs is ongoing.

Reply:
oh, so PHPStan has the `int|false` type for `strrpos`, but psalm does not have it yet ?

Reply:
Psalm definitely has it - @Ocramius is maybe talking about PHPStorm?

Reply:
Well, this has been inferred as `int|bool` here :|

Reply:
Maybe something else is happening? https://psalm.dev/r/d029859bdf works as expected

Reply:
Or possibly this was an  attempt to fix the possibly-false operandi issue (that you fixed with an explicit `(int)` cast below

Reply:
Keeping this as-is for now.